Archipelago.HollowKnight/Archipelago.cs

```diff
@@ -122,6 +120,15 @@ public void ConnectAndRandomize()
             ItemChangerMod.CreateSettingsProfile();
 
             ConnectToArchipelago();
+
+            goal = Goal.GetGoal(SlotOptions.Goal);
+            if (goal == null)
```

Comment:
As commented in the `Goals` file, have `Goal.GetGoal()` throw the exception. Then catch it out here in a `try..catch` block and log the error. Preferably, it would be best to gracefully dump the user to the main menu and provide feedback, but I'm partial for crashing the game here too. :)